### PR TITLE
Python correction

### PR DESCRIPTION
**Pull request by henriquedgarcia**
The python script has basicaly three errors:
- Processing must be done only in the luminance channel;
- The arrays used are of type UINT8 that do not represent the results of the operations correctly;
- The default options of the Sobel filter of the SciPy package do not detect the edges correctly.

Additionally I made some improvements:
- Cleaning the headers
- The right use of abstract class
- A pythonic way to do a getter.

### CODE REVIEW

File: python/siti.py [hunk 1]
```diff
@@ -70,7 +68,14 @@ def analyze_video(video):
...
     }
     i = 0
-    for frame in skvideo.io.vreader(video):
+    for frame in skvideo.io.vreader(video, as_grey=True):
+        if len(frame.shape) > 2:
+            width = frame.shape[-2]
+            height = frame.shape[-3]
+            frame = frame.reshape((height, width))
+
+        frame = frame.astype('float32')
```
Comment (slhck) [inline python/siti.py:77]: Could you maybe explain why this is needed? Just for curiosity. As there are no decimal values in the color range, an 8 bit integer should have been enough, or at least that was the understanding when implementing it.

**Review by slhck — COMMENTED** [review]
Looks good in general! I'm not in front of a computer for the next few days but I'll merge soon.

File: python/siti.py (hunk 1 above)
Comment (henriquedgarcia) [reply python/siti.py:77]: Yes, it is true, but to calculate the TI one frame is subtracted by another and the new array can have negative numbers.  To calculate the SI the Sobel filter returns an array of the same input type with values larger than 255. Then I change that to float32. In the opencv code this is also done in the [line 328 fom main.cpp code](https://github.com/Telecommunication-Telemedia-Assessment/SITI/blob/ca6e804934ee4842a4dcfd2ed17c783142fe0624/src/SITI/main.cpp#L328).

**Comment by slhck** [comment]
Thanks for your PR!

**Pull request merged**